src/streams/StreamScreen.js

```diff
@@ -22,22 +25,40 @@ class StreamScreen extends PureComponent<Props> {
     actions.navigateToEditStream(navigation.state.params.streamId);
   };
 
+  static contextTypes = {
+    styles: () => null,
+  };
+
+  toggleStreamPushNotification = () => {
+    const { subscriptions, navigation, auth } = this.props;
+    const { streamId } = navigation.state.params;
+    const subscription = subscriptions.find(x => x.stream_id === streamId) || NULL_SUBSCRIPTION;
+    toggleStreamNotification(auth, streamId, !subscription.push_notifications);
```

Comment:
Add an `s` to the name as `toggleStreamNotifications`?

Also, extract that to an action and retrieve the `auth` and `subscription` from the store there.